### PR TITLE
chore(observability): add tests to sinks for Data Volume tags

### PR DESCRIPTION
**Pull request by StephenWakely**
This PR adds tests for the Data Volume tags for the following sinks:

- [x] elasticsearch 
- [x] splunk_hec logs
- [x] splunk_hec metrics
- [x] loki 
- [x] new_relic
- [x] vector

The tests highlighted errors with the Vector sink, so there is also a fix for that sink to ensure the Data Volume tags (source, service) are included.

For the remaining sinks  (unless there is a pressing need for specific sinks) we will wait for the component validation framework to be implemented, which is a much cleaner way of testing for these tags. 

Reference: https://github.com/vectordotdev/vector/issues/17576

### CODE REVIEW

**Comment by jszwedko** [comment]
@StephenWakely is this ready for review? I know we discussed deferring other sinks in-lieu of the roll out of the new validation framework.

**Comment by StephenWakely** [comment]
> @StephenWakely is this ready for review? I know we discussed deferring other sinks in-lieu of the roll out of the new validation framework.

Yes!

File: lib/vector-common/src/request_metadata.rs [hunk 1]
```diff
@@ -145,6 +145,25 @@ impl GroupedCountByteSize {
...
+        }
+    }
+
+    /// Returns `true` if we are the `Untagged` variant - keeping a single count for all events.
+    #[must_use]
+    pub fn is_untagged(&self) -> bool {
+        match self {
+            GroupedCountByteSize::Tagged { .. } => false,
+            GroupedCountByteSize::Untagged { .. } => true,
+        }
```
Comment (neuronull) [inline lib/vector-common/src/request_metadata.rs:165]: **💭 thought:** any reason not to change that to


```suggestion
        !is_tagged()
```

?

File: src/sinks/splunk_hec/logs/integration_tests.rs [hunk 2]
```diff
@@ -145,6 +151,45 @@ async fn splunk_insert_message() {
...
+            tags: Tags {
+                emit_service: true,
+                emit_source: true,
+            },
+        },
+        true,
+    );
+}
+
+#[tokio::test]
```
[the remark this answers is not in the document]
Comment (StephenWakely) [reply src/sinks/splunk_hec/logs/integration_tests.rs:166]: This difference is `run_and_assert_data_volume_sink_compliance` vs `run_and_assert_sink_compliance`.

[the remark this answers is not in the document]
Comment (spencergilbert) [reply src/sinks/splunk_hec/logs/integration_tests.rs:166]: They're both calling `run_and_assert_data_volume_sink_compliance` though?

File: src/sinks/redis.rs [hunk 3]
```diff
@@ -126,7 +126,7 @@ pub struct RedisSinkConfig {
 
     /// The URL of the Redis endpoint to connect to.
     ///
-    /// The URL _must_ take the form of `protocol://server:port/db` where the protocol can either be
```
Comment (neuronull) [inline src/sinks/redis.rs]: was that intentional ?

**Review by neuronull — DISMISSED** [review]

File: src/sinks/splunk_hec/logs/integration_tests.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (StephenWakely) [reply src/sinks/splunk_hec/logs/integration_tests.rs:166]: When you say both what do you mean?

We have tests: `splunk_insert_message` and `splunk_insert_raw_message` that test without telemetry enabled - testing that the event and raw endpoints work. 

And we have `splunk_insert_message_data_volume` and `splunk_insert_raw_message_data_volume` that test with telemetry enabled - testing that the event and raw endpoints work and still emit the data volume tags.

File: src/sinks/redis.rs (hunk 3 above)
Comment (StephenWakely) [reply src/sinks/redis.rs]: No

File: src/sinks/splunk_hec/logs/integration_tests.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (spencergilbert) [reply src/sinks/splunk_hec/logs/integration_tests.rs:166]: Ah, _raw_ refers to the endpoint. I thought it had more to do with the event. Sorry for the confusion, just wanted to make sure there wasn't a copy+paste issue here.

**Review by spencergilbert — DISMISSED** [review]

**Review by neuronull — DISMISSED** [review]

**Review by neuronull — APPROVED** [review]

**Review by neuronull — APPROVED** [review]

**Pull request merged**